ocaml/xapi/helpers.ml

```diff
@@ -797,6 +797,13 @@ let is_pool_master ~__context ~host =
   let master_id = Db.Host.get_uuid ~__context ~self:master in
   host_id = master_id
 
+let assert_we_are_master ~__context =
+  if not (is_pool_master ~__context ~host:(get_localhost ~__context)) then
+    raise
+      Api_errors.(
+        Server_error (host_is_slave, [Pool_role.get_master_address ()])
```

Comment:
I think the error message is confusing but consistent with the code base: it reports a host is a slave but includes the address of the master.

Reply:
Yes, this function was moved from `xapi_psr.ml`, had a fix to report the address of current slave in the error.